src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -402,27 +387,16 @@ static void uart_driver_do_write(Context *ctx, term msg)
 
     free(buffer);
 
-    int local_pid = term_to_local_process_id(pid);
-    if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2) + REF_SIZE) != MEMORY_GC_OK)) {
```

Comment:
Removing this `memory_ensure_free` is not ok, since `port_send_reply` does not ensure memory. We need to keep it.
Please correct me if it is different.